src/main/java/com/bizo/awsstubs/services/dynamodb2/AmazonDynamoDBStub.java

```diff
@@ -728,6 +740,62 @@ public void addRows(Collection<Map<String, AttributeValue>> rows) {
       return matches;
     }
 
+    public List<Map<String, AttributeValue>> query(Map<String, Condition> keyConditions,
+      List<String> attributesToGet) {
+      List<Map<String, AttributeValue>> matches = new ArrayList<Map<String, AttributeValue>>();
+
+      List<Map<String, AttributeValue>> rows = this.getRows();
+
+      for (Map<String, AttributeValue> row : rows) {
+        boolean match = true;
+
+        for (Entry<String, Condition> keyCondition : keyConditions.entrySet()) {
+          String columnName = keyCondition.getKey();
+          Condition condition = keyCondition.getValue();
+
+          AttributeValue columnValue = row.get(columnName);
+
+          boolean comparisonMatch = comparisonMatch(columnValue, condition);
+          match &= comparisonMatch;
+        }
+
+        if (match) {
+          matches.add(row);
+        }
+      }
+
+      return matches;
+    }
+
+    private boolean comparisonMatch(AttributeValue columnValue, Condition condition) {
+      boolean comparisonMatch = false;
+
+      ComparisonOperator comparisonOperator = ComparisonOperator.fromValue(condition.getComparisonOperator());
+      AttributeValue comparisonAttributeValue = condition.getAttributeValueList().get(0);
+
+      switch (comparisonOperator) {
+        case EQ:
+          comparisonMatch = AttributeValueComparator.equal(columnValue, comparisonAttributeValue);
+          break;
```

Comment:
Instead of having the comparisonMatch local variable, and a break line in each case, you could do:

```
case EQ:
  return AttributeValueComparator.equal(columnValue, comparisonAttributeValue);
```

Also minor but I might shorten `comparisonAttributeValue` to just `comparisonValue`.